src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -148,6 +150,152 @@ public function getVariation(Experiment $experiment, $userId, $attributes = null
   }
 
   /**
+   * Get the variation the user is bucketed into for the given FeatureFlag
+   * @param  FeatureFlag $featureFlag The feature flag the user wants to access
+   * @param  string      $userId      user id
+   * @param  array       $attributes  user attributes
+   * @return array/null  {"experiment" : Experiment, "variation": Variation } / null
+   */
+  public function getVariationForFeature(FeatureFlag $featureFlag, $userId, $attributes){
+    //Evaluate in this order: 
+    //1. Attempt to bucket user into all experiments in the feature flag.
+    //2. Attempt to bucket user into rollout in the feature flag.
+
+    // Check if the feature flag is under an experiment and the the user is bucketed into one of these experiments
+    $result = $this->getVariationForFeatureExperiment($featureFlag, $userId, $attributes);
```

Comment:
conceptually this is wrong getVariationForFeatureExperiment has the same attributes we have in this method. You should iterate experiment and pass experiment one by one. and return variation if you find.

Reply:
Make better separation. Regarding your method args concern,
even in Java, getVariationForFeatureRollout has FeatureFlag as argument and not Rollout. And I have added a lot of unit tests, refactoring this will need a lot of effort